lib/internal/Magento/Framework/Api/Search/SearchCriteriaBuilder.php

```diff
@@ -83,8 +88,10 @@ public function addFilter(\Magento\Framework\Api\Filter $filter)
      */
     public function addSortOrder($field, $direction)
     {
-        $this->sortOrderBuilder->setDirection($direction)
-            ->setField($field);
+        $sortOrder = $this->sortOrderBuilder->setDirection($direction)
+            ->setField($field)
+            ->create();
+        $this->sortOrders = [$sortOrder];
```

Comment:
When you multiple times will call the addSortOrder method, it will replace sort orders. I think it's not the expected result.
In stead, I think we should add a new sort order to the sortOrders array. 
What do you think?

Also, I think we should cover it with unit or integration test. Could you please do so?

Reply:
I only had it replacing sort orders as that is what it was doing previously. If adding multiple sort orders is desired we can update it to do so.

Reply:
I've changed the `addSortOrder` method to add multiple sort orders, and added a `setSortOrders` method to replace the sort orders in the builder, which is the same functionality as the regular SearchCriteriaBuilder.